compiler-core/src/build/module_loader.rs

```diff
@@ -73,6 +73,11 @@ where
             if meta.fingerprint != SourceFingerprint::new(&source_module.code) {
                 tracing::debug!(?name, "cache_stale");
                 return Ok(Input::New(source_module));
+            } else if self.mode == Mode::Lsp {
```

Comment:
Do you think we only want to do this in the instance that we are in LSP mode and the previous cache was for an invalid state? To avoid things like changing git branches performing excess compilation?

Likely not the most impactful optimisation today seeing as Gleam codebases are not huge, and this only happens when lots of files are changed, but it might be simple to add. I'm not sure myself.

Reply:
I think it would be doable to explicitly store "these modules had partial compilation occur so ignore the fingerprint" rather than always ignoring the fingerprint for the lsp. Just need to wire up some plumbing to store/pass that info down and reset it on successful compile

Reply:
Think I've implemented something that does what we want for this. lmk what you think